src/entities/BlockNode/BlockNode.spec.ts

```diff
@@ -137,9 +130,228 @@ describe('BlockNode', () => {
+        data: {
+          object: {
+            nestedObject: { value: 'value' },
+          },
+        },
+        parent: {} as EditorDocument,
+      });
+
+      blockNode.serialized;
+
+      expect(spy)
+        .toHaveBeenCalledTimes(1);
+    });
+
+    it('should call .serialized getter of TextNode in a nested object', () => {
+      const spy = jest.spyOn(TextNode.prototype, 'serialized', 'get');
+      const blockNode = new BlockNode({
+        name: createBlockToolName('paragraph'),
+        data: {
+          object: {
+            nestedObject: {
+              text: {
+                $t: BlockChildType.Text,
+                value: '',
+                fragments: [],
+              },
+            },
+          },
+        },
+        parent: {} as EditorDocument,
+      });
+
+      blockNode.serialized;
+
+      expect(spy)
+        .toHaveBeenCalledTimes(1);
+    });
+
+    it('should call .serialized getter of ValueNodes in an array inside an object', () => {
+      const spy = jest.spyOn(ValueNode.prototype, 'serialized', 'get');
+      const blockNode = new BlockNode({
+        name: createBlockToolName('paragraph'),
+        data: {
+          object: {
+            array: [ 'value' ],
+          },
+        },
+        parent: {} as EditorDocument,
+      });
+
+      blockNode.serialized;
+
+      expect(spy)
+        .toHaveBeenCalledTimes(1);
+    });
+
+    it('should call .serialized getter of TextNode in an array inside an object', () => {
+      const spy = jest.spyOn(TextNode.prototype, 'serialized', 'get');
+      const blockNode = new BlockNode({
+        name: createBlockToolName('paragraph'),
+        data: {
+          object: {
+            array: [ {
+              text: {
+                $t: BlockChildType.Text,
+                value: '',
+                fragments: [],
+              },
+            } ],
+          },
+        },
+        parent: {} as EditorDocument,
+      });
+
+      blockNode.serialized;
+
+      expect(spy)
+        .toHaveBeenCalledTimes(1);
+    });
+
+    it('should call .serialized getter of ValueNodes in an object inside an array', () => {
+      const spy = jest.spyOn(ValueNode.prototype, 'serialized', 'get');
+      const blockNode = new BlockNode({
+        name: createBlockToolName('paragraph'),
+        data: {
+          array: [ { value: 'value' } ],
+        },
+        parent: {} as EditorDocument,
       });
+
+      blockNode.serialized;
+
+      expect(spy)
+        .toHaveBeenCalledTimes(1);
+    });
+
+    it('should call .serialized getter of TextNode in an object inside an array', () => {
+      const spy = jest.spyOn(TextNode.prototype, 'serialized', 'get');
+      const blockNode = new BlockNode({
+        name: createBlockToolName('paragraph'),
+        data: {
+          array: [ {
+            object: {
+              text: {
+                $t: BlockChildType.Text,
+                value: '',
+                fragments: [],
+              },
```

Comment:
I thought we decided to detect text nodes by value and fragment fields. Is $t field required here?

Reply:
I think it would be more consistent to add $t property for text nodes as well. So in case we'd like to change the data structure of a text node, we could still detect it using $t prop